db/scripts/init_tables.sql

```diff
@@ -95,11 +95,22 @@ ALTER TABLE domains ADD COLUMN IF NOT EXISTS queue_weeks INTEGER DEFAULT 4;
 ALTER TABLE domains ADD COLUMN IF NOT EXISTS testing_start TIMESTAMP;
 
 -- Drop & re-add constraint
-ALTER TABLE domains DROP CONSTRAINT domains_pkey;
-ALTER TABLE domains ADD PRIMARY KEY (domain, status);
+BEGIN;
+    ALTER TABLE domains DROP CONSTRAINT domains_pkey;
+    ALTER TABLE domains ADD PRIMARY KEY (domain, status);
+COMMIT;
 
 ALTER TABLE IF EXISTS aggregated_scans DROP COLUMN IF EXISTS connected;
 ALTER TABLE IF EXISTS aggregated_scans ADD COLUMN IF NOT EXISTS with_mxs INTEGER DEFAULT 0;
 
 ALTER TABLE domains ADD COLUMN IF NOT EXISTS mta_sts BOOLEAN DEFAULT FALSE;
 
+DO $$
+BEGIN
+    IF NOT EXISTS (SELECT conname
+                   FROM   pg_constraint
+                   WHERE  conname = 'aggregated_scans_time_source_key')
+    THEN
+        ALTER TABLE aggregated_scans ADD UNIQUE (time, source);
+    END IF;
+END$$;
```

Comment:
We could stick to the txn if we add a line that adds it on the first run as is done for domains [here](https://github.com/EFForg/starttls-backend/pull/232/files#diff-d4f05f3fe241a72ae233cc0a4441a75dR40) -- wdyt?

Reply:
ooh yeah that's great!